server/lib/packets/packet.go

```diff
@@ -92,16 +103,39 @@ func (pkt *RecvPacket) Process(t Transport) {
 	}
 }
 
+func Actor2ProtoError(aerr *yue.ActorError) *proto.Err {
+	if aerr.Is(yue.ActorUnavailable) {
+		return &proto.Err {
+			Type: proto.Error_TemporaryUnavailable,
+		}
+	} else {
+		return &proto.Err {
+			Type: proto.Error_RuntimeError,
+			Explanation: aerr.Error(),			
+		}
+	}
+}
+
 //SendError sents error with specified reason
-func SendError(src Source, msgid uint32, reason proto.Error_Type) {
-	log.Printf("SendError:%v", reason);
+func SendError(src Source, msgid uint32, err error) {
+	perr, ok := err.(*proto.Err)
+	if !ok {
+		aerr, ok2 := err.(*yue.ActorError)
```

Comment:
__[gometalinter]__
* [gotype] undeclared name: yue
* [gotype] err (variable of type error) cannot have dynamic type *invalid type (missing method Error)